lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -3573,12 +3573,11 @@ bool ExprAvailabilityWalker::diagnoseDeclRefAvailability(
   return false;
 }
 
-/// Diagnose uses of API annotated '@unavailableFromAsync' when used from
-/// asynchronous contexts.
-/// Returns true if a diagnostic was emitted, false otherwise.
+/// Diagnose misuses of asynchronous API in asynchronous contexts.
```

Comment:
This is diagnosing both now right 🤔? Ie. use of a sync API in async context when there's an async alternative as well as use of an unavailable async API?

Reply:
Yep

Reply:
I'm not sure that's clear to me from `asynchronous API in asynchronous context`, though I suppose you could argue that you're misusing the asynchronous API by using the synchronous API 🤷

Reply:
Something like that. I can change the comment if that's what you're asking for.

Reply:
Sure, could even just drop the first asynchronous really. Or expand it. Up to you.

Reply:
Done